lib/plausible_web/controllers/auth_controller.ex

```diff
@@ -561,6 +561,39 @@ defmodule PlausibleWeb.AuthController do
     |> redirect(to: redirect_to)
   end
 
+  def google_auth_callback(conn, %{"error" => error, "state" => state} = params) do
+    [site_id, _redirect_to] = Jason.decode!(state)
+    site = Repo.get(Plausible.Site, site_id)
+
+    case error do
+      "access_denied" ->
+        conn
+        |> put_flash(
+          :error,
+          "We were unable to authenticate your Google Analytics account. Please check that you have granted us permission to 'See and download your Google Analytics data' and try again."
+        )
+        |> redirect(to: Routes.site_path(conn, :settings_general, site.domain))
+
+      message when message in ["server_error", "temporarily_unavailable"] ->
+        conn
+        |> put_flash(
+          :error,
+          "We are unable to authenticate your Google Analytics account at the moment. The authentication service is currently unavailable. Please try again in a few moments."
```

Comment:
```suggestion
          "We are unable to authenticate your Google Analytics account because Google's authentication service is temporarily unavailable. Please try again in a few moments."
```

Trying to clarify that:
1. The issue is on Google's side, not us
2. Should be a temporary thing, trying again should work

Reply:
Good one!